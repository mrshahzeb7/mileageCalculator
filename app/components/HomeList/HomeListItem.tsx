import React from 'react';
import {
  StyleProp,
  StyleSheet,
  Text,
  TextStyle,
  View,
  ViewStyle,
} from 'react-native';
import {HomeSectionItem} from '../../screens/Home';
import {Icon} from '../Icon/Icon';
import {borderRadius, colors, iconSizes, spacing} from '../../theme';

interface Props {
  item: HomeSectionItem;
  index: number;
}

export const HomeListItem: React.FC<Props> = ({item, index}) => {
  const $container: StyleProp<ViewStyle> = [
    styles.container,
    index === 0 && styles.topRadius,
    item?.style,
  ];
  return (
    <View style={$container}>
      {!!item?.title && <Text style={styles.title}>{item?.title}</Text>}
      <View style={styles.dataWrapper}>
        <View style={styles.valueWrapper}>
          {!!item?.leftIcon && (
            <Icon
              size={iconSizes.tiny}
              style={styles.leftIcon}
              icon={item?.leftIcon}
            />
          )}
          {!!item?.sign && <Text style={styles.sign}>{item?.sign}</Text>}
          {!!item?.value && <Text style={styles.value}>{item?.value}</Text>}
          {!!item?.unit && <Text style={styles.unit}>{item?.unit}</Text>}
        </View>
        <Text style={styles.details}>{item?.details}</Text>
      </View>
    </View>
  );
};

const $row: ViewStyle = {
  flexDirection: 'row',
  alignItems: 'center',
};

const textSpacing: TextStyle = {
  marginLeft: spacing.horizontal.tiny,
};

const radius = borderRadius.medium;

// styles
const styles = StyleSheet.create({
  container: {
    backgroundColor: colors.primary,
    paddingHorizontal: spacing.horizontal.medium,
    paddingVertical: spacing.vertical.tiny * 1.1,
  },
  dataWrapper: {
    justifyContent: 'space-between',

    ...$row,
  },
  valueWrapper: {
    ...$row,
  },
  leftIcon: {
    marginRight: spacing.horizontal.medium,
  },
  title: {
    color: colors.textDim,
    marginBottom: spacing.vertical.tiny,
  },
  sign: {
    color: colors.text,
  },
  value: {
    color: colors.text,
  },
  unit: {
    color: colors.textDim,
    ...textSpacing,
  },
  details: {
    color: colors.textDim,
  },

  topRadius: {
    borderTopLeftRadius: radius,
    borderTopRightRadius: radius,
  },
});
