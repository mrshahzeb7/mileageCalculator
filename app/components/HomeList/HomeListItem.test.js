import 'react-native';
import React from 'react';
import {HomeListItem} from './HomeListItem';
import renderer from 'react-test-renderer';

describe('HomeListItem', () => {
  it(' HomeListItem renders correctly', () => {
    const tree = renderer.create(<HomeListItem />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
