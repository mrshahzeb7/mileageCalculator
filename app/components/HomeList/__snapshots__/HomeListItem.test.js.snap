// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeListItem  HomeListItem renders correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#2e2f32",
        "paddingHorizontal": 37.5,
        "paddingVertical": 14.674000000000001,
      },
      false,
      undefined,
    ]
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    />
    <Text
      style={
        {
          "color": "#6a6f71",
        }
      }
    />
  </View>
</View>
`;
